static/src/javascripts/projects/common/modules/ui/bannerPicker.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import breakingNews from 'common/modules/onward/breaking-news';
+import optInEngagementAlert from 'common/modules/identity/global/opt-in-engagement-banner.js';
+
+type Banner = {
+    canShow: () => Promise<boolean>,
+    show: () => void,
+};
+
+let banners: Array<Banner> = [breakingNews, optInEngagementAlert];
+
+let results: Array<'pending' | boolean> = new Array(banners.length).fill(
+    'pending',
+    0
+);
```

Comment:
Could we initialise these variables in `init`? Then we could avoid the side effects of loading this module, and we wouldn't need to expose a function just for testing

Reply:
`banners` and `results` sit outside the scope of `init` as `resetChecks` needs to be able to reset their values when this module is under test, `getSuccessfulBannerIndex` could go inside `init` though.

Reply:
You could still define the variables in the top scope, but then use `init` to initialise them?

Reply:
OK I see what you meant, yes I can do this!

Reply:
On second thought, if I initialise them in `init` it'll override the values whenever I run `init` from the test module.

Reply:
Good point. It sort of indicates that the banners shouldn't to be imported in this module. Perhaps they should be passed from the place where `bannerPicker.init` is called (`enhanced/common`). That would keep this module purer, make it more testable, and pass the banner-loading back up to the naturally more imperative `enhanced/common` module.